src/controllers/legacycontrollermapping.cpp

```diff
@@ -0,0 +1,66 @@
+#include "controllers/legacycontrollermapping.h"
+
+void LegacyControllerMapping::loadSettings(const QFileInfo& mappingFile,
+        UserSettingsPointer pConfig,
+        const QString& controllerName) const {
+    QString controllerKey = QString(CONTROLLER_SETTINGS_PREFERENCE_GROUP_KEY)
```

Comment:
Some users use their setup on two devices. 
This way we need to maintain the exact path on both.
Can you make that relative to the mixxx.cfg file or the controller folder? 

What is the use case when tweaking a shipped mapping and then decide to move it to ./Mixxx/controller? I guess it would be nice to maintain the settings. 

How does the combo box handle two mappings with the same name? If only one is selectable anyway, we can also use that as key. 

Do we need to consider file encoding? 
What happen if we have non ASCII characters in the mapping file path. Does it play nice?

Reply:
> I guess it would be nice to maintain the settings

I would argue it would be best not to clone the existing settings: If one goes to  change the behavior of a script, then importing a set of setting initially design for a script which has since been altered may likely cause issues.
I know this is a common practice to copy the original script and tweak the settings defined in there, but this feature is specifically replacing the need for that, meaning that the only use-case that will remain is for users that want to rework or fork a script, and so IMHO, they should work with stock settings.

> How does the combo box handle two mappings with the same name?

User-defined will always be shown first with a different icons, and stock one second, after a separator. I'm not sure how it looks like if you have two user-defined (or system-defined) with the same name tho.

> Do we need to consider file encoding?

I believe QString should cover that for us for file contents, when it comes to title, I would need to check that, but I would assume this is likely to be OS or FS specific.

Reply:
I mean, If you copy your `~/.mixxx` folder to a different location lets say `~/.mixxx2` and than start Mixxx via 
`mixxx --settings_path ~/.mixxx2` 
I think we should allow this use case.

Reply:
I see, that makes sense. With the latest commit, I have added Setting path and Resource path substitution so that should achieve the behavior you are requesting.